src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
+    }
 
+    private void ensureCapacity() {
+        if (size == elementData.length) {
+            int newCapacity = elementData.length * 2;
+            Object[] newArray = new Object[newCapacity];
+            System.arraycopy(elementData, 0, newArray, 0, elementData.length);
+            elementData = newArray;
+        }
```

Comment:
The `ensureCapacity` method effectively doubles the size of the array when it runs out of space. This is a good strategy, but consider extracting the resize factor as a constant to make it clear and easily adjustable.